unitTests/controllers/QueryControllerTest.ts

```diff
@@ -101,6 +101,54 @@ export function QueryControllerTest() {
       }, 10);
     });
 
+    describe('closeModalBox', () => {
+      let modalBoxCloseSpy: jasmine.Spy;
+      beforeEach(() => {
+        modalBoxCloseSpy = jasmine.createSpy('modal');
+        test.cmp.modalBox = {
+          close: modalBoxCloseSpy
+        };
+      });
+
+      afterEach(() => {
+        modalBoxCloseSpy = null;
+        test.cmp.modalBox = null;
+        delete test.cmp;
+        delete test.env;
+      });
+
+      it('should close the modals if the closeModalBox is set to true in the executeQuery options', () => {
+        test.cmp.executeQuery({
+          closeModalBox: true
+        });
+        expect(modalBoxCloseSpy).toHaveBeenCalled();
+      });
+
+      it('should not close the modals if the closeModalBox is set to false in the executeQuery options', () => {
+        test.cmp.executeQuery({
+          closeModalBox: false
+        });
+        expect(modalBoxCloseSpy).not.toHaveBeenCalled();
+      });
+
+      it('should close the modals if the closeModalBox is set to true', () => {
+        test.cmp.closeModalBox = true;
+        test.cmp.executeQuery();
+        expect(modalBoxCloseSpy).toHaveBeenCalled();
+      });
+
+      it('should not close the modals if the closeModalBox is set to false', () => {
+        test.cmp.closeModalBox = false;
+        test.cmp.executeQuery();
+        expect(modalBoxCloseSpy).not.toHaveBeenCalled();
+      });
+
+      it('should close the modals if the closeModalBox is unset', () => {
```

Comment:
I think a better test description would be `by default` instead of `if the closeModalBox is unset`. So it would become:

`it('should close the modals by default')`